File: sample-conference-videochat/sample-conference-videochat/Components/NewDialogScreen/Views/UserTableViewCell.m
//
//  UserTableViewCell.m
//  sample-conference-videochat
//
//  Created by Injoit on 04.02.2020.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "UserTableViewCell.h"
#import "UILabel+Chat.h"
#import "UIView+Chat.h"

@implementation UserTableViewCell
#pragma mark - Life Cycle
- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.userAvatarImageView.hidden = YES;
    self.userAvatarLabel.hidden = NO;
    [self.userAvatarLabel setRoundedLabelWithCornerRadius:20.0f];
    self.contentView.backgroundColor = UIColor.clearColor;
    self.checkBoxView.backgroundColor = UIColor.clearColor;
    [self.checkBoxView setRoundBorderEdgeColorView:4.0f borderWidth:1.0f color:nil borderColor:[UIColor colorWithRed:0.42f green:0.48f blue:0.57f alpha:1.0f]];
    self.muteButton.hidden = YES;
    [self.muteButton setImage:[UIImage imageNamed:@"mute_opponent"] forState:UIControlStateNormal];
    [self.muteButton setImage:[UIImage imageNamed:@"unmute_opponent"] forState:UIControlStateSelected];
    self.muteButton.selected = NO;
}

#pragma mark - Setup
- (void)setUserColor:(UIColor *)userColor {
    _userColor = userColor;
    self.userAvatarLabel.backgroundColor = self.userColor;
}

- (IBAction)didTapMuteButton:(UIButton *)sender {
    sender.selected = !sender.selected;
    if (self.didPressMuteButton) {
        self.didPressMuteButton(sender.selected);
    }
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];
    
    if (self.isSelected) {
        self.checkBoxImageView.hidden = NO;
        self.contentView.backgroundColor = [UIColor colorWithRed:0.85f green:0.89f blue:0.97f alpha:1.0f];
        [self.checkBoxView setRoundBorderEdgeColorView:4.0f
                                           borderWidth:1.0f
                                                 color:[UIColor colorWithRed:0.22f green:0.47f blue:0.99f alpha:1.0f]
                                           borderColor:[UIColor colorWithRed:0.22f green:0.47f blue:0.99f alpha:1.0f]];
        
    } else {
        self.checkBoxImageView.hidden = YES;
        self.contentView.backgroundColor = UIColor.clearColor;
        [self.checkBoxView setRoundBorderEdgeColorView:4.0f
                                           borderWidth:1.0f
                                                 color:UIColor.clearColor
                                           borderColor:[UIColor colorWithRed:0.42f green:0.48f blue:0.57f alpha:1.0f]];
    }
    self.userAvatarLabel.backgroundColor = self.userColor;
}

- (void)setHighlighted:(BOOL)highlighted {
    [super setHighlighted:highlighted];
    
    if (self.isHighlighted) {
        self.contentView.backgroundColor = [UIColor colorWithRed:0.85f green:0.89f blue:0.97f alpha:1.0f];
        [self.checkBoxView setRoundBorderEdgeColorView:4.0f
                                           borderWidth:1.0f
                                                 color:[UIColor colorWithRed:0.22f green:0.47f blue:0.99f alpha:1.0f]
                                           borderColor:[UIColor colorWithRed:0.22f green:0.47f blue:0.99f alpha:1.0f]];
        
    } else {
        self.contentView.backgroundColor = UIColor.clearColor;
        [self.checkBoxView setRoundBorderEdgeColorView:4.0f
                                           borderWidth:1.0f
                                                 color:UIColor.clearColor
                                           borderColor:[UIColor colorWithRed:0.42f green:0.48f blue:0.57f alpha:1.0f]];
    }
    self.userAvatarLabel.backgroundColor = self.userColor;
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/Components/NewDialogScreen/UserTableViewCell.swift
//
//  UserTableViewCell.swift
//  sample-conference-videochat-swift
//
//  Created by Injoit on 12/26/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

//protocol UserCellDelegate: class {
//    func didTapMuteButton(_ userCell: UserTableViewCell)
//}

struct UserCellConstant {
    static let reuseIdentifier =  "UserTableViewCell"
}

class UserTableViewCell: UITableViewCell {
    
    //MARK: - Properties
//    weak var delegate: UserCellDelegate?
    
    @IBOutlet weak var userAvatarImageView: UIImageView!
    @IBOutlet weak var userAvatarLabel: UILabel!
    @IBOutlet weak var userNameLabel: UILabel!
    @IBOutlet weak var checkBoxView: UIView!
    @IBOutlet weak var checkBoxImageView: UIImageView!
    @IBOutlet weak var muteButton: UIButton!
    
    /**
     *  Mute user block action.
     */
    var didPressMuteButton: ((_ isMuted: Bool) -> Void)?
    
    var userColor: UIColor? {
        didSet {
            userAvatarLabel.backgroundColor = userColor
        }
    }
    
    let unmutedImage = UIImage(named: "mute_opponent")!
    let mutedImage = UIImage(named: "unmute_opponent")!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        userAvatarImageView.isHidden = true
        userAvatarLabel.isHidden = false
        muteButton.isHidden = true
        muteButton.setImage(unmutedImage, for: .normal)
        muteButton.setImage(mutedImage, for: .selected)
        muteButton.isSelected = false
        userAvatarLabel.setRoundedLabel(cornerRadius: 20.0)
        contentView.backgroundColor = .clear
        checkBoxView.backgroundColor = .clear
        checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                                 borderWidth: 1.0,
                                                 borderColor: UIColor(red:0.42, green:0.48, blue:0.57, alpha:1))
    }
    
    @IBAction func didTapMuteButton(_ sender: UIButton) {
        sender.isSelected = !sender.isSelected
        didPressMuteButton?(sender.isSelected)
//        delegate?.didTapMuteButton(self)
    }
    
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        
        if self.isSelected == true {
            contentView.backgroundColor = UIColor(red:0.85, green:0.89, blue:0.97, alpha:1)
            
            checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                                     borderWidth: 1.0,
                                                     color: UIColor(red:0.22, green:0.47, blue:0.99, alpha:1),
                                                     borderColor: UIColor(red:0.22, green:0.47, blue:0.99, alpha:1))
        } else {
            contentView.backgroundColor = .clear
            
            checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                                     borderWidth: 1.0,
                                                     color: .clear,
                                                     borderColor: UIColor(red:0.42, green:0.48, blue:0.57, alpha:1))
        }
        userAvatarLabel.backgroundColor = userColor
    }
    
    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
        super.setHighlighted(highlighted, animated: animated)
        
        if self.isHighlighted == true {
            contentView.backgroundColor = UIColor(red:0.85, green:0.89, blue:0.97, alpha:1)
            checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                                     borderWidth: 1.0,
                                                     color: UIColor(red:0.22, green:0.47, blue:0.99, alpha:1),
                                                     borderColor: UIColor(red:0.22, green:0.47, blue:0.99, alpha:1))
        } else {
            contentView.backgroundColor = .clear
            checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                                     borderWidth: 1.0,
                                                     color: .clear,
                                                     borderColor: UIColor(red:0.42, green:0.48, blue:0.57, alpha:1))
        }
        userAvatarLabel.backgroundColor = userColor
    }
}
